docs/specs/native_histograms.md

```diff
@@ -1771,14 +1771,18 @@ The following describes all the operations that actually _do_ work.
 Addition (`+`) and subtraction (`-`) work between two compatible histograms.
 These operators add or subtract all matching bucket populations and the count
 and the sum of observations. Missing buckets are assumed to be empty and
-treated accordingly. Subtraction might result in negative histograms, see
-[notes above](#unary-minus-and-negative-histograms). Generally, both operands
-should be gauges. Adding and subtracting counter histograms requires caution,
-but PromQL allows it. Adding a gauge histogram and a counter histogram results
-in a gauge histogram. Adding two counter histograms with contradicting counter
-reset hints triggers a warn-level annotation. (TODO: The latter not yet
-implemented. Also, subtraction doesn't check/modify counter reset hints yet.
-This should be documented in detail in the PromQL docs.)
+treated accordingly. Generally, both operands should be gauges. Adding and
+subtracting counter histograms requires caution, but PromQL allows it. Adding a
+gauge histogram and a counter histogram results in a gauge histogram. Adding
+two counter histograms results in a counter histogram. If the two operands
+share the same counter reset hint, the resulting counter histogram retains that
+counter reset hint. Otherwise, the resulting counter reset hint is set to
+`UnknownCounterReset`. The result of a subtraction is always marked as a gauge
+histogram because it might result in negative histograms, see [notes
+above](#unary-minus-and-negative-histograms). In any case, adding or
```

Comment:
```suggestion
above](#unary-minus-and-negative-histograms). Adding or
```

I'm also wondering if we should keep this line, as there is no user-visible warning until [reset merge algorithm](https://github.com/prometheus/prometheus/issues/15346) is resolved.